server/routing/cleanup_tasks.go

```diff
@@ -46,19 +45,34 @@ func (f *remoteLabelFilter) Filter(e query.Entry) bool {
 }
 
 // CleanupManager handles all background cleanup and republishing tasks for the routing system.
-// This includes label republishing to DHT, stale remote label cleanup, and orphaned record cleanup.
+// This includes CID provider republishing, GossipSub label republishing, stale remote label cleanup, and orphaned record cleanup.
 type CleanupManager struct {
-	dstore   types.Datastore
-	storeAPI types.StoreAPI
-	server   *p2p.Server
+	dstore      types.Datastore
+	storeAPI    types.StoreAPI
+	server      *p2p.Server
+	publishFunc func(context.Context, string, *corev1.Record) error // Publishing callback (captures routeRemote state)
 }
 
 // NewCleanupManager creates a new cleanup manager with the required dependencies.
-func NewCleanupManager(dstore types.Datastore, storeAPI types.StoreAPI, server *p2p.Server) *CleanupManager {
+// The publishFunc is injected from routeRemote.publishToNetwork to avoid circular dependencies
+// while still providing access to DHT and GossipSub publishing logic.
+//
+// Parameters:
+//   - dstore: Datastore for label storage
+//   - storeAPI: Store API for record operations
+//   - server: P2P server for DHT operations
+//   - publishFunc: Callback for publishing (from routeRemote.publishToNetwork)
+func NewCleanupManager(
+	dstore types.Datastore,
+	storeAPI types.StoreAPI,
+	server *p2p.Server,
+	publishFunc func(context.Context, string, *corev1.Record) error,
```

Comment:
this function type can be defined in `server/routing/types` such as

```go
type PublishEventHandler func(...) (...)
```

Reply:
see above